featuretools/primitives/standard/binary_transform.py

```diff
@@ -280,6 +280,7 @@ def equal(x_vals, y_vals):
                 categories = set(x_vals.cat.categories).union(set(y_vals.cat.categories))
                 x_vals = x_vals.cat.add_categories(categories.difference(set(x_vals.cat.categories)))
                 y_vals = y_vals.cat.add_categories(categories.difference(set(y_vals.cat.categories)))
+                y_vals = y_vals.cat.reorder_categories(x_vals.cat.categories)
```

Comment:
Certain comparisons were resulting in an error because the categories were not ordered the same between `x_vals` and `y_vals` so added this line here and below to fix.

Reply:
should we make an issue to revisit this to figure why it changed / how costly this reorder is?

Reply:
Sure. Created #1584 for this.